dlc-wallet/src/main/resources/postgresql/dlc/migration/V1__dlc_db_baseline.sql

```diff
@@ -1,108 +1,138 @@
-CREATE TABLE "wallet_dlcs"
+CREATE TABLE "global_dlc_data"
```

Comment:
So it seems that at minimum the contract parameters should be moved out of this table. 

Here is the data model @rorp used for the contract explorer, i think you can crib quite a bit from that?

https://gitlab.com/suredbits/suredbits-api/-/commit/a3e15528a0f56ad454e9fc9d6ad4c0ea70a1e499#14ce9ef498223ef8e08f948a62f7e3af8b7c018d_0_1

Reply:
I guess i think this table should be renamed to `contract_info` or something like that. It seems like that is what this table represents? 

https://github.com/bitcoin-s/bitcoin-s/blob/394e9f773d3a73751c02629ee774337e8b70c508/core/src/main/scala/org/bitcoins/core/protocol/tlv/TLV.scala#L1221

If that is the case, we should have this table foreign keys on the sub tables that are used to compose this table (tables that represent `oracleInfo` and `contractDescriptor` in `ContractInfo`). I believe things like `oracle_threshold`, `oracle_params`. 

Something just smells about this table where we have three different domains having data directly inserted in here (not just ids for fk)

1. Oracle information
2. Contract information
3. wallet information (`key_index`, `change_index`, `txid`s etc)

Reply:
Yeah so we currently split up the oracle info into other tables by having the announcements point back, as for the contract descriptor it is not as easy as the enum vs numeric are very different and there isn't much it can be split up by.

We could move these
```
    "oracle_threshold"      INTEGER NOT NULL, 
    "oracle_params"         TEXT,
    "contract_descriptor"   TEXT    NOT NULL,
    "contract_maturity"     INTEGER NOT NULL,
    "contract_timeout"      INTEGER NOT NULL,
    "total_collateral"      INTEGER NOT NULL, 
    "fee_rate"              TEXT    NOT NULL,
    "fund_output_serial_id" INTEGER NOT NULL,
```

to a separate table, however this will just be another database call we need to do.

Reply:
I think it separates the concerns better IMO. From all the information in that table, can we eventually support specific `enum_contract_descriptor` and `numeric_contract_descriptor` tables that fk back into the table you have laid out in your comment?

Reply:
If wanted to later do it, would require some complicated migrations, I don't think we would need that though, the descriptors are pretty compact and we don't really have a reason to query for individualized pieces.

Reply:
Ok that is fine with me